apps/src/lib/config/ethereum_bridge/params.rs

```diff
@@ -0,0 +1,106 @@
+//! Blockchain-level parameters for the configuration of the Ethereum bridge.
+use serde::{Deserialize, Serialize};
+
+/// Represents a configuration value for an Ethereum address e.g.
+/// 0x6B175474E89094C44Da98b954EedeAC495271d0F
+#[derive(Clone, Eq, PartialEq, Debug, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Address(String);
+
+/// Represents a configuration value for the minimum number of
+/// confirmations an Ethereum event must reach before it can be acted on.
+#[derive(Clone, Copy, Eq, PartialEq, Debug, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct MinimumConfirmations(u64);
+
+impl Default for MinimumConfirmations {
+    fn default() -> Self {
+        Self(100)
+    }
+}
+
+/// Represents chain parameters for the Ethereum bridge.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Config {
+    /// Minimum number of confirmations needed to trust an Ethereum branch.
+    /// This must be at least one.
+    pub min_confirmations: MinimumConfirmations,
+    /// The addresses of the Ethereum contracts that need to be directly known
+    /// by validators.
+    pub contracts: Contracts,
+}
+
+/// Represents all the Ethereum contracts that need to be directly know about by
+/// validators.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Contracts {
+    /// The Ethereum address of the ERC20 contract that represents this chain's
+    /// native token.
+    pub native_erc20: Address,
+    /// The Ethereum address of the bridge contract.
+    pub bridge: UpgradeableContract,
+    /// The Ethereum address of the governance contract.
+    pub governance: UpgradeableContract,
+}
+
+/// Represents a configuration value for the version of a contract that can be
+/// upgraded. Starts from 1.
+#[derive(Clone, Copy, Eq, PartialEq, Debug, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct ContractVersion(u64);
```

Comment:
```suggestion
pub struct ContractVersion(NonZeroU64);
```
same thing here, since you want the version to start counting from 1